pallets/ethereum-token-transfers/src/lib.rs

```diff
@@ -258,5 +285,35 @@ pub mod pallet {
 
             Ok(())
         }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight(T::WeightInfo::add_tip_v2())]
+        pub fn add_tip(
+            origin: OriginFor<T>,
+            message_id: MessageId,
+            amount: u128,
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+
+            T::Currency::transfer(
```

Comment:
I'm not sure I like this, maybe we should be a bit more modular. perhaps move this out to TipHandler impl?

Reply:
https://github.com/moondance-labs/tanssi/pull/1332/commits/8fafbab5fc2d9c6313927c92bdfd6f97a643097d